particle warning

```
/ssd/aspect/source/particle/integrator/rk_2.cc:43:60: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
         Assert(std::distance(begin_particle, end_particle) == old_velocities.size(),
                                                            ^
```
(same in euler and rk4)

--- a/source/particle/integrator/euler.cc
+++ b/source/particle/integrator/euler.cc
@@ -38,7 +38,7 @@ namespace aspect
                                        const std::vector<Tensor<1,dim> > &,
                                        const double dt)
       {
-        Assert(std::distance(begin_particle, end_particle) == old_velocities.size(),
+        Assert(static_cast<unsigned int> (std::distance(begin_particle, end_particle)) == old_velocities.size(),
                ExcMessage("The particle integrator expects the velocity vector to be of equal size "
                           "to the number of particles to advect. For some unknown reason they are different, "
                           "most likely something went wrong in the calling function."));

--- a/source/particle/integrator/rk_2.cc
+++ b/source/particle/integrator/rk_2.cc
@@ -40,7 +40,7 @@ namespace aspect
                                      const std::vector<Tensor<1,dim> > &velocities,
                                      const double dt)
       {
-        Assert(std::distance(begin_particle, end_particle) == old_velocities.size(),
+        Assert(static_cast<unsigned int> (std::distance(begin_particle, end_particle)) == old_velocities.size(),
                ExcMessage("The particle integrator expects the old velocity vector to be of equal size "
                           "to the number of particles to advect. For some unknown reason they are different, "
                           "most likely something went wrong in the calling function."));

--- a/source/particle/integrator/rk_4.cc
+++ b/source/particle/integrator/rk_4.cc
@@ -40,7 +40,7 @@ namespace aspect
                                      const std::vector<Tensor<1,dim> > &velocities,
                                      const double dt)
       {
-        Assert(std::distance(begin_particle, end_particle) == old_velocities.size(),
+        Assert(static_cast<unsigned int> (std::distance(begin_particle, end_particle)) == old_velocities.size(),
                ExcMessage("The particle integrator expects the old velocity vector to be of equal size "
                           "to the number of particles to advect. For some unknown reason they are different, "
                           "most likely something went wrong in the calling function."));

